docs/Virt-v2v-wrapper.md

```diff
@@ -92,6 +92,132 @@ All keys in this section are subject to change without notice and should not be
 
 * `rhv_debug`: optional key to enable debugging of `ovirtsdk4` API calls.
 
+OpenStack instance migration is supported, but only to another OpenStack cloud.
+OpenStack VMs cannot currently be migrated to RHV, for example. To set up a
+migration from OpenStack to OpenStack, the source and destination clouds must
+be configured with conversion host instances as shown in the sub-topics below.
+The migration works by attaching volumes to the source and destination
+conversion hosts and transferring data from inside the conversion hosts over
+SSH.
+
+### Source OpenStack cloud
+
+* A conversion host instance must be launched from the UCI, in the same project
```

Comment:
It would be problematic to make the wrapper UCI agnostic, because some entity needs to induce the source wrapper to export the block devices, and therefore needs to know how to start the wrapper. Right now the destination wrapper only knows how to start the v2v-conversion-host container the way it is packaged in the UCI. If it isn't allowed to do that, then there is some complexity that needs to be pushed back to the caller (meaning ansible or CloudForms) to start the export on the source and provide the destination wrapper with a mapping between OpenStack volumes and NBD URLs.

Here are the only other options I have thought of so far:
- Remove exports.py and run nbdkit directly on the source VM (not in the wrapper container)
- Move all of the OpenStack source host stuff out of the wrapper into the os-migrate ansible playbook that is going to be responsible for transferring everything else (like flavors and security groups)
- Wait for the planned wrapper API so the destination wrapper just needs a URL to the source API and can do what it needs to do without caring how either side was launched

I guess I should just document the UCI for now, but I would more than happy to remove the podman invocation contortions from the wrapper itself. The question of where this should go is maybe better decided on a mailing list, so I will start writing something up.

Reply:
No problem, no need for anything drastic liked that. Even though it could be nice to have it uci-agnostic, it's enough to just document what UCI is (add a link or something). If someone starts working on this later on, it would be very confusing. Imagine I documented pre-copy using qemu-internal terms that you cannot even look up online. It would be impossible for someone to continue or contribute.

Reply:
I have now added some text about the UCI, and I added a new input argument so the caller can specify the right container ID.